paasta_tools/cli/cmds/status.py

```diff
@@ -713,13 +715,37 @@ def status_kubernetes_job_human(
         )
 
 
+def _strip_config_prefix(config_sha: str) -> str:
+    prefix = "config"
+    prefix_len = len(prefix)
+    if len(config_sha) > prefix_len and config_sha[:prefix_len] == prefix:
+        return config_sha[prefix_len:]
+    return config_sha
+
+
 def print_flink_status(
-    cluster: str, service: str, instance: str, output: List[str], status, verbose: int
+    cluster: str,
+    service: str,
+    instance: str,
+    output: List[str],
+    status,
+    metadata,
+    verbose: int,
 ) -> int:
     if status is None:
         output.append(PaastaColors.red("    Flink cluster is not available yet"))
         return 1
 
+    config_sha = None
```

Comment:
```
if metadata and metadata.labels:
    config_sha = metadata.labels.get("yelp.com/paasta_config_sha")
    if config_sha:
        if config_sha.startswith("config"):
            config_sha = config_sha[5:]
        output.append(f"    Config SHA: {config_sha}")
```
?

Reply:
yes, that's one way to simplify this. I am also considering to always show config sha, and when not known then show "(unknown)"

Reply:
yeah, I think I'd rather show 'unknown'. It's a situation that shouldn't really ever happen. If anything, it's worth outputting something like `config: ERROR - Could not determine config sha`.

Reply:
well, that error is not really contributing much. I will just show "(unknown)"

Reply:
It'll signal to the user that there is something wrong. It should be a rare case, but I'd rather do something to let the user know that this isn't an expected state. You could even just error out at this point

Reply:
The small issue I have with letting the error out is that you could hypothetically run `paasta status` against a cluster running an old version of paasta-tools, which does not yet set the metadata. Also, if the current version of paasta-tools is installed but there is a bug somewhere else (e.g. preventing the config_sha label from being set in the metadata), I think being able to show the rest of the state will aid troubleshooting.

Reply:
> you could hypothetically run paasta status against a cluster running an old version of paasta-tools, which does not yet set the metadata

I don't think you need to worry about this: we have control over the versions of paasta tools we run everywhere.

>  if the current version of paasta-tools is installed but there is a bug somewhere else (e.g. preventing the config_sha label from being set in the metadata), I think being able to show the rest of the state will aid troubleshooting.

If that label doesn't exist, then the cluster isn't going to be functional anyway (the operator depends on it). I think it'd be a pretty exceptional circumstance and we should treat is as such

Reply:
ok, I just removed the defensive checks so if this errors out, the user will get all the Python stack he needs to learn about the problem.